website/content/en/docs/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,178 @@
+
+Particularly, note the following changes:
+
+1. The naming convention of the reconciler struct may differ. For example, in case of memcached operator the reconciler struct in SDK project is mentioned as [`ReconcileMemcached`][sdk_reconcile_struct] whereas in Kubebuilder project it would be `MemcachedReconciler`.
+
+```go
+type MemcachedReconciler struct {
+	Log    logr.Logger
+	Client client.Client
+	Scheme *runtime.Scheme
+}
+```
+
+2. Update the import aliases in the `Reconcile()` function and copy over any remaining helper functions from `pkg/controller/<kind>/<kind>_controller.go` to `api/<version>/<kind>_controller.go`.
+
+Refer to [`memcached_controller.go`][kb_memcached_controller] for the example implementation of `Reconcile()` logic which would be present in `api/v1alpha1/memcached_controller.go` of Memcached operator.
+
+## Generate CRDs
+
+Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/crd/bases` folder.
+
+```sh
+$ make manifests
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+```
+
+## Operator Manifests
+
+### Operator deployment manifests
+
+The operator deployment manifest [`deploy/operator.yaml`][deployment_yaml] from the old project should be copied over to `config/manager/manager.yaml` in the new project.
+
+**Note**: 
+The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the current operator.yaml of sdk project.
+> **// TODO:** Explain label propagation for metrics collection.
+
+### RBAC permissions
+
+The RBAC manifests present in `config/rbac/` of a Kubebuilder project are generated through the [RBAC markers][rbac_markers] present in `controller/<kind>_controller.go`. They can be added as comments above the `Reconcile()` method.
+
+In our example of memcached operator, add the following rbac markers in `memcached_controller.go`:
+
+```Go
+
+...
+// +kubebuilder:rbac:groups=apps,resources=deployments;pods;daemonsets;replicasets;statefulsets,verbs=get;update;patch;list;create;delete;watch
+
+func (r *MemcachedReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	reqLogger := log.WithValues("Request.Namespace", req.Namespace, "Request.Name", req.Name)
+	reqLogger.Info("Reconciling Memcached")
+
+...
+```
+**Note**:
+To update `config/rbac/role.yaml` after changing the markers, run `make manifests`.
+
+The project can now be built and the operator can be deployed on cluster. For further steps regarding the deployment of operator, creation of custom resource and cleaning up of resources, refer to [quickstart guide][kb_quickstart].
+
+
```

Comment:
Is not missing how to move the main.go impl and add the sdk as dep to allow users still using the features provided by it? Are not the users using the metrics and or adding new schemas for example?

Reply:
The operator scope doc is still in progress I think: https://github.com/operator-framework/operator-sdk/blob/master/website/content/en/docs/kubebuilder/quickstart.md#operator-scope
Will direct the users there once done.

Reply:
Regards the scope:

SDK does not work as Kubebuilder. By default, all projects scaffolded by Kubebuilder will be cluster escaped and not namespaced scoped as it is in SDK. So, IMO we need to clarify that and described what are the steps to keep the project with namespaced scope. 

 Regards the sdk dep:

SDK provides features which users can have been using in their projects so IMO we need to clarify how to keep this feature as well. E.g test-framework, k8sutils and etc.

Reply:
Hi @hasbro17 and @joelanford, 

Wdyt? Is it something that we could skip in the first moment?

Reply:
@camilamacedo86 I think if we try to cover everything in the migration guide it will turn into a duplicate of the user-guide and advanced topics.
We just want the basic migration steps in this guide to be able to move over the apis/controllers and manifests.

And then for users who wan to enable `WATCH_NAMESPACE ` and scope their operator correctly we'll have to point them to the operator-scope doc (WIP) to make the necessary changes.

So I think we can leave that out for now.